_posts/2021-01-18-changelog-139.markdown

```diff
@@ -0,0 +1,113 @@
+* Incrementally update experimental [local crates index](https://github.com/intellij-rust/intellij-rust/pull/6390).
+  This speeds up the experimental `Cargo.toml` dependency completion. In the future, this would affect other new code insight based on the local crates index as well.
+  Note, the corresponding feature is disabled by default for now.
+  You can enable it via `org.rust.crates.local.index` option in `Experimental Feature` dialog (`Help | Find Action` and type `Experimental features` to open the dialog)
+
+## Fixes
+
+* [New name resolution](https://github.com/intellij-rust/intellij-rust/issues/6217) engine improvements:
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6576 -->
+    * Support `#![macro_use]` attribute at file level
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6543 -->
+    * Consider [textual scoping](https://doc.rust-lang.org/reference/macros-by-example.html#textual-scope) when resolving macros
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6568 -->
+    * Properly process items with restricted visibility that contains an absolute path, e.g. `pub(in crate::inner1)`
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6566 -->
+    * Fix name resolution of qualified macro calls inside items in some cases
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6545 -->
+    * Properly handle renaming of enum variants
+
+  Note, you can enable the new engine via `Use experimental name resolution engine` option in `Preferences | Languages & Frameworks | Rust` settings
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6606 -->
+* Fix name resolution and completion of `Self` keyword for nested structs, enums, and `impl` blocks
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6613 -->
+* Improve the new `Fill function arguments` quick-fix (by [@Kobzol])
+    * Support filling arguments of a closure call
+    * Take actual types into account when constructing default values of generic types
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6515 -->
+* `Remove parameter` quick-fix no longer removes references to the modified function (by [@Kobzol])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6574 -->
+* Fix resolution of Cargo features for aliased packages like `foo_alias = { package = "foo", path = "../foo"}`
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6588 -->
+* Fix `Trait implementation` inspection on `impl` blocks containing macro-expanded members
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6541 -->
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6542 -->
+* Improve parser recovery and syntax error messages for incomplete paths
+
+## Internal Improvements
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6578 -->
+* Introduce `TomlVisitor`. This class would be used to implement inspections that process PSI elements of a `Cargo.toml` file (by [@Kobzol])
```

Comment:
```suggestion
* Introduce `TomlVisitor`. This class can be used to implement inspections in TOML files in a simple and efficient way (by [@Kobzol])
```

because this class can be used for any inspection in any TOML file, not only for `Cargo.toml`